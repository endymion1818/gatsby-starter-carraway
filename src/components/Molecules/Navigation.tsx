import Link from 'gatsby-link'
import React, { FC } from 'react'

export interface INavigationProps {
  navitems: Array<{
    title: string
    url: string
    navOrder: number
  }>
}

const Navigation: FC<INavigationProps> = ({ navitems }) => {
  return (
    <nav>
      <ul>
        {navitems.map(item => (
          <li key={item.title}>
            <Link to={item.url}>{item.title}</Link>
          </li>
        ))}
      </ul>
    </nav>
  )
}

export default Navigation
