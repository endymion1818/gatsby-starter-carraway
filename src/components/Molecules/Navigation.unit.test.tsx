import { cleanup, render } from '@testing-library/react'
import React from 'react'
import Navigation from './Navigation'

const defaultProps = {
  navitems: [
    {
      title: 'test',
      url: 'some-url',
      navOrder: 1,
    },
  ],
}

describe('<Navigation />', () => {
  beforeEach(cleanup)
  it('Should render', () => {
    const { container } = render(<Navigation {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })
})
