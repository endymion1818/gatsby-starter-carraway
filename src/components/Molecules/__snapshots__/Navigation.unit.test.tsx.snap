// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Should render 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <a
          href="/some-url"
        >
          test
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
